did-utils/src/crypto/utils.rs

```diff
@@ -17,12 +28,25 @@ pub fn generate_seed(initial_seed: &[u8]) -> Result<[u8; BYTES_LENGTH_32], &str>
     Ok(seed)
 }
 
-/// clone the content of the slice into a new array
+/// Clones the content of the slice into a new array.
+///
 /// It is important to clone the data, as we don't want key material to be hazardously modified.
-pub fn clone_slice_to_array(slice: &[u8; BYTES_LENGTH_32]) -> [u8; BYTES_LENGTH_32] {
-    // Create a new array of the expected length
-    let mut array: [u8; BYTES_LENGTH_32] = [0; BYTES_LENGTH_32];
-    // clone the data from the slice into the array
+///
+/// # Arguments
+///
+/// * `slice` - The slice to clone.
+///
+/// # Returns
+///
+/// A new array containing the cloned data.
+///
+/// # Panics
+///
+/// Panics if the length of the slice is not equal to `BYTES_LENGTH_32`.
```

Comment:
My bad. that is a mistake. I haven't correctly checked that the input parameter is always 32 bytes long. By the way the panics warning was mentioned because the called function `clone_from_slice` may panics if the 2 slices have different lengths.